plugins/inputs/qbittorrent/qbittorrent.go

```diff
@@ -0,0 +1,228 @@
+package qbittorrent
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+var globalMainData *MainData
+
+type QBittorrent struct {
+	Host     string        `toml:"host"`
+	Port     int           `toml:"port"`
```

Comment:
Maybe better to have 1 field `url` (or `urls` to add support for multiple endpoints), then you can also drop the `tls` field.